assets/javascripts/discourse/components/upcoming-events-list.gjs

```diff
@@ -68,6 +67,31 @@ export default class UpcomingEventsList extends Component {
     return this.router.currentRoute.attributes?.category?.id;
   }
 
+  get title() {
+    const categorySlug = this.router.currentRoute.attributes?.category?.slug;
+    const titleSetting = this.siteSettings.map_events_title;
+
+    if (titleSetting === "") {
+      return I18n.t("discourse_post_event.upcoming_events_list.title");
+    }
+
+    const categories = JSON.parse(titleSetting).map(
+      ({ category_slug }) => category_slug
+    );
+
+    if (categories.includes(categorySlug)) {
+      const titleMap = JSON.parse(titleSetting);
+      const customTitleLookup = titleMap.find(
+        (o) => o.category_slug === categorySlug
+      );
+      const custom_title = customTitleLookup?.custom_title;
+
+      return custom_title;
```

Comment:
shorter and with no underscore_case variable:

```suggestion
      return customTitleLookup?.custom_title;
```

Reply:
ope, totally missed that. thank you.